smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -190,6 +213,42 @@ contract MixinPurchase is
     }
   }
 
+  /**
+  * Renew a given token
+  * @notice only works for non-free, expiring, ERC20 locks
+  * @param _tokenId the ID fo the token to renew
+  * @param _referrer the address of the person to be granted UDT
+  */
+  function renewMembershipFor(
+    uint _tokenId,
+    address _referrer
+  ) public {
+    _onlyIfAlive();
+    _lockIsUpToDate();
+    _isKey(_tokenId);
+
+    // check the lock
+    require(_originalDurations[_tokenId] != type(uint).max, 'NON_EXPIRING_LOCK');
+    require(_originalPrices[_tokenId] != 0, 'FREE_LOCK');
```

Comment:
I think we should enable free locks to be automatically renewed too

Reply:
Yes it was some misconceptions about free locks not needed to be renewed but thats not necessarily the case